lib/blocks.php

```diff
@@ -590,3 +590,27 @@ function gutenberg_multiple_block_styles( $metadata ) {
 	return $metadata;
 }
 add_filter( 'block_type_metadata', 'gutenberg_multiple_block_styles' );
+
+/**
+ * Styles should be loaded in the head for block themes.
+ *
+ * For classic ones we need to enqueue them to the body
+ * because the wp_head action (and wp_enqueue_scripts)
+ * happens before the render_block.
+ *
+ * See https://core.trac.wordpress.org/ticket/53494.
+ *
+ * @param string $style HTML content to load in the <head> or <body>.
+ */
+function gutenberg_enqueue_block_support( $style ) {
```

Comment:
The function wp_add_inline_style, which is similar receives a styles string and outputs it and it receives the styles without "style" tag so I guess this function should also not include the style tag.

On another question would it make sense to use wp_add_inline_style in this function instead of wp_enqueue_scripts? So for example a plugin can remove our styles etc...

Reply:
> The function wp_add_inline_style, which is similar receives a styles string and outputs it and it receives the styles without "style" tag so I guess this function should also not include the style tag.

Can do that.

> On another question would it make sense to use wp_add_inline_style in this function instead of wp_enqueue_scripts? So for example a plugin can remove our styles etc...

I'm on the fence here. My first instinct is to do not change how this works because I don't know why we made it this way in the first place. This PR just fixes the issue with block themes enqueuing the block supports in the body when they should use the head.

Reply:
https://github.com/WordPress/gutenberg/pull/38750/commits/3ed7cd07e6cf9211745d012f7cad43d8c9a00853 consumers no longer need to provide the `<style>` tag.

Reply:
It would be nice to have a reserved block supports style handle. That doesn't output the `link` element but can be used for inline styles.

Maybe it's something we can explore for 6.0.